index.js

```diff
@@ -1,24 +1,46 @@
-let body = document.querySelector("#body")
-let h1 = document.createElement("h1")
-body.append(h1)
-h1.innerText = "Random Star Wars Ship Generator"
-let button = document.createElement("button")
-button.id = "button"
-body.append(button)
-button.innerText = "GET ME A SHIP!"
-let ul = document.createElement("ul")
-body.append(ul)
-let p = document.createElement("p")
-ul.append(p)
-
-button.addEventListener("click", function(evt) {
-    fetch("https:/swapi.dev/api/starships/")
-    .then (res => res.json())
-    .then (function(starship){
-        function randomNumber() {
-            let i = Math.floor(Math.random() * (10))
-            return i
-        }
-        p.innerText = starship.results[randomNumber()].name
-    })
-})
\ No newline at end of file
+function randomShipName(ships) {
+	const index = Math.floor(Math.random() * (ships.length)),
+		ship = ships[index];
+
+	if (ship)
+		return ship.name;
+	else
+		return 'Uh oh! Ship not found :[';
+}
+
+const body = document.querySelector('body'),
+	h1 = document.createElement('h1'),
+	button = document.createElement('button'),
+	p = document.createElement('p');
+
+let ships;
+
+// page title
+h1.innerText = 'Random Star Wars Ship Generator';
+
+body.append(h1);
+
+// CTA
+button.innerText = 'GET ME A SHIP!';
+
+button.disabled = true;
+
+body.append(button);
+
+// ship container
+body.append(p);
+
+// get ships
+fetch('https:/swapi.dev/api/starships/')
+	.then(res => res.json())
+	.then(res => {
+		ships = starship.results;
+
+		if (ships && ships.length)
+			button.removeAttribute('disabled');
+	});
```

Comment:
I separated the button press from the fetch. Instead you can fetch all the ships upon page load, then simply grab a new one upon button press. You should notice this increases the responsiveness of the button. Obviously, if you expected the content of the ships API to change between button presses, this approach would not work.